crates/client/rpc/src/rpc_methods/get_transaction_receipt.rs

```diff
@@ -0,0 +1,555 @@
+        .sierra_classes()
+        .get_sierra_class(class_hash)
+        .map_err(|e| {
+            log::error!("Failed to fetch sierra class with hash {class_hash}: {e}");
+            StarknetRpcApiError::InternalServerError
+        })?
+        .ok_or_else(|| {
+            log::error!("The sierra class with hash {class_hash} is not present in db backend");
+            StarknetRpcApiError::InternalServerError
+        })?;
+
+    let contract_class = mp_transactions::utils::sierra_to_casm_contract_class(contract_class).map_err(|e| {
+        log::error!("Failed to convert the SierraContractClass to CasmContractClass: {e}");
+        StarknetRpcApiError::InternalServerError
+    })?;
+
+    let contract_class = ContractClassBf::V1(ContractClassV1Bf::try_from(contract_class).map_err(|e| {
+        log::error!("Failed to convert the compiler CasmContractClass to blockifier CasmContractClass: {e}");
+        StarknetRpcApiError::InternalServerError
+    })?);
+
+    Ok(UserOrL1HandlerTransaction::User(UserTransaction::Declare(declare_tx, contract_class)))
+}
+
+fn tx_l1_handler<A, B, BE, G, C, P, H>(
+    client: &Starknet<A, B, BE, G, C, P, H>,
+    chain_id: Felt,
+    block_number: u64,
+    l1_handler: mp_transactions::HandleL1MessageTransaction,
+) -> RpcResult<UserOrL1HandlerTransaction>
+where
+    A: ChainApi<Block = B> + 'static,
+    B: BlockT,
+    P: TransactionPool<Block = B> + 'static,
+    BE: Backend<B> + 'static,
+    C: HeaderBackend<B> + BlockBackend<B> + StorageProvider<B, BE> + 'static,
+    C: ProvideRuntimeApi<B>,
+    C::Api: StarknetRuntimeApi<B> + ConvertTransactionRuntimeApi<B>,
+    G: GenesisProvider + Send + Sync + 'static,
+    H: HasherT + Send + Sync + 'static,
+{
+    let chain_id = chain_id.0.into();
+    let tx_hash = l1_handler.compute_hash::<H>(chain_id, false, Some(block_number));
+    let paid_fee =
+        client.backend.l1_handler_paid_fee().get_fee_paid_for_l1_handler_tx(tx_hash.into()).map_err(|e| {
+            log::error!("Failed to retrieve fee paid on l1 for tx with hash `{tx_hash:?}`: {e}");
+            StarknetRpcApiError::InternalServerError
+        })?;
+
+    Ok(UserOrL1HandlerTransaction::L1Handler(l1_handler, paid_fee))
+}
+
+fn execution_infos<A, B, BE, G, C, P, H>(
+    client: &Starknet<A, B, BE, G, C, P, H>,
+    previous_block_hash: B::Hash,
+    transactions: Vec<UserOrL1HandlerTransaction>,
+) -> RpcResult<TransactionExecutionInfo>
+where
+    A: ChainApi<Block = B> + 'static,
+    B: BlockT,
+    P: TransactionPool<Block = B> + 'static,
+    BE: Backend<B> + 'static,
+    C: HeaderBackend<B> + BlockBackend<B> + StorageProvider<B, BE> + 'static,
+    C: ProvideRuntimeApi<B>,
+    C::Api: StarknetRuntimeApi<B> + ConvertTransactionRuntimeApi<B>,
+    G: GenesisProvider + Send + Sync + 'static,
+    H: HasherT + Send + Sync + 'static,
+{
+    let execution_infos = client
+		.client
+		.runtime_api()
+		.re_execute_transactions(previous_block_hash, vec![], transactions.clone())
```

Comment:
you can modify the parameters like this to retrieve only the information from the last transaction
```rust
re_re_execute_transactions(previous_block_hash, <previous_tx>, <last_tx>)
```
and remove `pop()`